tests/config_tests.ml

```diff
@@ -0,0 +1,224 @@
+open OUnit2
+open Lib.Config.ParseConfig
+open Lib.Config.Formats
+
+let check_ok f fname i =
+  let description =
+    Printf.sprintf
+      "%s is Ok on well-formed input"
+      fname
+  in
+  let test _ = i |> f |> Result.is_ok |> assert_bool "actually not Ok" in
+  description >:: test
+
+let wf =
+  [ ("source_type"  , "application/pdf"   ) ;
+    ("target_type"  , "application/pdf"   ) ;
+    ("shell_command", "soffice-to-pdfa.sh") ;
+  ]
+
+let extra = ("extra", "extra") :: wf
+
+let nwf =
+  [ ("source_type"  , "application/pdf"   ) ;
+    ("target_type!" , "application/pdf"   ) ;
+    ("shell_command", "soffice-to-pdfa.sh") ;
+  ]
+
+let wf_to_entry_ok    = check_ok entry_of_assoc "entry_of_assoc" wf
+let extra_to_entry_ok = check_ok entry_of_assoc "entry_of_assoc" extra
+
+let check_error f fname i =
+  let description =
+    Printf.sprintf
+      "%s is Error on ill-formed input"
+      fname
+  in
+  let test _ = i |> f |> Result.is_error |> assert_bool "actually not Error" in
+  description >:: test
+
+let nwf_to_entry_error = check_error entry_of_assoc "entry_of_assoc" nwf
+
+let check_entry l description st tt ss =
+  let l_parsed   = Result.get_ok (entry_of_assoc l)       in
```

Comment:
Would we be able to get away with not using `get_ok`?  For example, if we did this, wouldn't that make `check_entry` total rather than partial?  

```ocaml
let check_entry l description st tt ss =
  let l_parsed   = entry_of_assoc l       in
  let st_check _ = assert_equal l_parsed.source_type   (Ok st) in
  let tt_check _ = assert_equal l_parsed.target_type   (Ok tt) in
  let ss_check _ = assert_equal l_parsed.shell_command (Ok ss) in
```

Or is there an implicit invariant in here guaranteeing an `Ok` result?

Reply:
It's definitely true that `check_entry` is partial. My justification is that checking entries is preceded by a `check_ok`. This makes it possible to determine if a parse was bad because it returned an Error or because it returned an Ok but the data was bad. It might be possible to combine these two checks into a single test group, but I think at the possible expense of simplicity.

Reply:
Ah, got it!  Thanks.  I had a feeling there'd be an explanation on these lines when I raised the question.